### PR TITLE
ref: Start typing eventstream

### PR DESCRIPTION
**Pull request by lynnagara**
We are about to rewrite the post process forwarder into a streaming consumer using arroyo. Let's start typing the eventstream to avoid introducing errors during the refactor.

### CODE REVIEW

**Review by wedamija — APPROVED** [review]
Thanks!

jfyi, my team is also doing work on the post process forwarder in the next 2-3 weeks for performance issues. We will be changing the structure of the messages in some way. Since transactions can have multiple issues associated with them, we will need to have `is_new` etc associated with each group. We haven't decided on what this looks like, but just a heads up that we will also be working here.

File: src/sentry/eventstream/base.py [hunk 1]
```diff
@@ -1,6 +1,8 @@
 import logging
-from typing import Literal, Optional, Union
+from datetime import datetime
+from typing import Any, Collection, Literal, Mapping, Optional, Sequence, Union
 
+from sentry.eventstore.models import BaseEvent
```
Comment (wedamija) [inline src/sentry/eventstream/base.py:5]: Might need to import this `if TYPE_CHECKING`

Comment (asottile-sentry) [reply src/sentry/eventstream/base.py]: and to add `from __future__ import annotations` to the top of the file -- https://youtu.be/zH_F2xC0LOk

Comment (wedamija) [reply src/sentry/eventstream/base.py]: I've wondered a few times if we should just bulk import that in every file

Comment (asottile-sentry) [reply src/sentry/eventstream/base.py]: I'm in favor of it -- there's a few quirks with things that (ab)use runtime typing like pydantic / etc. that we'd want to make sure we wouldn't anger

might throw together a branch to see how angy it is

Comment (lynnagara) [reply src/sentry/eventstream/base.py]: Thanks

**Pull request merged**